packages/issuer/src/blockchain-facade/CertificationRequest.ts

```diff
@@ -108,51 +113,11 @@ export class CertificationRequest extends PreciseProofEntity implements ICertifi
     }
 
     async sync(): Promise<CertificationRequest> {
-        const { issuer } = this.configuration
-            .blockchainProperties as Configuration.BlockchainProperties<Registry, Issuer>;
-
-        const issueRequest = await issuer.getCertificationRequest(this.id);
-        const decodedData = await issuer.decodeData(issueRequest.data);
-
-        this.owner = issueRequest.owner;
-        this.fromTime = Number(decodedData['0']);
-        this.toTime = Number(decodedData['1']);
-        this.deviceId = decodedData['2'];
-        this.approved = issueRequest.approved;
-        this.revoked = issueRequest.revoked;
-
-        const certificationRequestedLogs = await getEventsFromContract(
-            issuer,
-            issuer.filters.CertificationRequested(null, this.id, null)
-        );
-
-        const creationBlock = await issuer.provider.getBlock(
-            certificationRequestedLogs[0].blockNumber
+        const offChainData = await this.configuration.offChainDataSource.certificateClient.getCertificationRequest(
+            this.id
```

Comment:
Is there a case for origin where this would be better read from the blockchain?

Reply:
Not sure. This used to be the case. Reading it from the backend is more secure, albeit more centralized.

Reply:
Reading from the blockchain would be required for on-chain issuance, like without external issuer. This good for now 👍